src/backend/sql/gnc-address-sql.cpp

```diff
@@ -60,33 +60,32 @@ static GncSqlColumnTableEntry col_table[] =
     { "phone", CT_STRING, ADDRESS_MAX_PHONE_LEN,        COL_NNUL, "phone" },
     { "fax",   CT_STRING, ADDRESS_MAX_FAX_LEN,          COL_NNUL, "fax" },
     { "email", CT_STRING, ADDRESS_MAX_EMAIL_LEN,        COL_NNUL, "email" },
-    { NULL }
-};
+});
 
 typedef void (*AddressSetterFunc) (gpointer, GncAddress*);
 typedef GncAddress* (*AddressGetterFunc) (const gpointer);
 
 static void
 load_address (const GncSqlBackend* be, GncSqlRow* row,
               QofSetterFunc setter, gpointer pObject,
-              const GncSqlColumnTableEntry* table_row)
+              const GncSqlColumnTableEntry& table_row)
 {
     const GValue* val;
     gchar* buf;
     GncAddress* addr;
     AddressSetterFunc a_setter = (AddressSetterFunc)setter;
-    const GncSqlColumnTableEntry* subtable;
     const gchar* s;
 
+
     g_return_if_fail (be != NULL);
     g_return_if_fail (row != NULL);
     g_return_if_fail (pObject != NULL);
-    g_return_if_fail (table_row != NULL);
 
     addr = gncAddressCreate (be->book, QOF_INSTANCE(pObject));
-    for (subtable = col_table; subtable->col_name != NULL; subtable++)
+    for (auto const& subtable_row : col_table)
```

Comment:
auto would get me a copy of the GncSqlColumnTableEntry and I need a const reference to it instead for polymorphism.